lib/nolibc/musl-imported/include/time.h

```diff
@@ -42,6 +37,26 @@ struct tm {
 	const char *__tm_zone;
 };
 
+typedef long time_t;
```

Comment:
Unfortunately, declaring the types directly here in this file break other headers, like `stat.h` that also depend on the declaration. The thing is that libC's `stat.h` should not include `time.h` by itself because this defines more than a user actually expects with inluding `stat.h` (would be bad practice). What you need to do is to move these declarations to nolibc's `shareddef.h` and guard each declaration like `useconds_t`:
```c
#if defined(__NEED_useconds_t) && !defined(__DEFINED_useconds_t)
typedef unsigned useconds_t;
#define __DEFINED_useconds_t
#endif
```
In the `time.h` header you include the type definitons with:
```c
#define __NEED_suseconds_t
#define __NEED_struct_timeval
#define __NEED_struct_timespec
#define __NEED_timer_t
#define __NEED_clockid_t
#define __NEED_clock_t

#include <nolibc-internal/shareddefs.h>
```

Reply:
Thank you for the comments. I adapted the code so that the compilation succeeds.